src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -52,51 +52,57 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
             _channels = new RedisChannels(typeof(THub).FullName);
             _protocol = new RedisProtocol(hubProtocolResolver.AllProtocols);
 
-            var writer = new LoggerTextWriter(logger);
             RedisLog.ConnectingToEndpoints(_logger, options.Value.Options.EndPoints, _serverName);
-            _redisServerConnection = _options.Connect(writer);
+        }
 
-            _redisServerConnection.ConnectionRestored += (_, e) =>
+        private async Task EnsureRedisServerConnection()
+        {
+            if (_redisServerConnection == null)
             {
+                var writer = new LoggerTextWriter(_logger);
```

Comment:
Need a lock. The lifetime manager can be called from multiple concurrent tasks.

Reply:
(semaphore lock since it is async)